### ISSUE TITLE
`yarn rw setup auth supabase` installs the wrong version of @redwoodjs/auth

### ISSUE THREAD

**Issue by Tobbe** [redwoodjs/redwood#2696]
This is related to #2685. But I don't know if solving one will also solve the other, so I'm creating a new issue.

After changing auth providers I noticed some problems with rendering pages wrapped in `<Private>` tags. The URL would update, but the page was just blank.

That's when I noticed this in my `git diff`

```diff
diff --git a/web/package.json b/web/package.json
index 18ee063..db56b01 100644
--- a/web/package.json
+++ b/web/package.json
@@ -18,10 +18,11 @@
     "@fortawesome/free-solid-svg-icons": "^5.15.2",
     "@fortawesome/react-fontawesome": "^0.1.14",
     "@headlessui/react": "^0.2.0",
-    "@redwoodjs/auth": "0.29.0",
+    "@redwoodjs/auth": "^0.32.2",
     "@redwoodjs/forms": "0.29.0",
     "@redwoodjs/router": "0.29.0",
     "@redwoodjs/web": "0.29.0",
+    "@supabase/supabase-js": "^1.12.0",
     "@tailwindcss/typography": "^0.2.0",
     "autoprefixer": "^10.0.2",
     "chart.js": "^2.9.4",
```

And my `yarn rw info` is this

```
  System:
    OS: Windows 10 10.0.19041
  Binaries:
    Node: 15.4.0 - ~\AppData\Local\Temp\yarn--1622613089525-0.014593816711588437\node.CMD
    Yarn: 1.22.4 - ~\AppData\Local\Temp\yarn--1622613089525-0.014593816711588437\yarn.CMD
  Browsers:
    Chrome: 90.0.4430.212
    Edge: Spartan (44.19041.964.0), Chromium (91.0.864.37)
  npmPackages:
    @redwoodjs/core: 0.29.0 => 0.29.0
```

So I got the wrong version of @redwoodjs/auth. I think running `yarn rw setup auth` should install the version of @redwoodjs/auth that matches the user's version of @redwoodjs/core (0.29.0 in my case)

**Comment by dthyresson** [comment]
Looking at the setup command:

```js
      {
        title: 'Adding required web packages...',
        task: async () => {
          if (!isProviderSupported(provider)) {
            throw new Error(`Unknown auth provider '${provider}'`)
          }
          await execa('yarn', [
            'workspace',
            'web',
            'add',
            ...providerData.webPackages,
            '@redwoodjs/auth',
          ])
        },
      },
```

You ned to 

* know current version of redwood 
* use this to set `'@redwoodjs/auth@0.your.version',


Not sure how to find current RW version, but perhaps get Project from structure can provide?

**Comment by thedavidprice** [comment]
The command `yarn rw info` uses `envinfo` under the hood, but that might be overly cumbersome. I know the structure package has a helper — just be careful as it's likely to be deprecated (prevent regression via unit test?).

Otherwise, just parse the root package.json would be my suggestion (also note: in the v0.34 release, we'll be replacing @redwoodjs/core with @redwoodjs/cli as the only package in root).

**Comment by dthyresson** [comment]
> just parse the root package.json would be my suggestion

Of course - that makes perfect sense.

**Comment by Tobbe** [comment]
Fixed by #3885
